# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1412883950 14400
#      Thu Oct 09 15:45:50 2014 -0400
# Branch gui-release
# Node ID eb9b2160e87807a9b41cc54cbad1a19e31de50db
# Parent  eee862631953693a6cb133839e33db8d9d39b92d
fix build problem with --enable-64 (bug #43319)

* QtHandlesUtils.cc (makeImageFromCData): Ensure arguments to qMin
have the same type.

diff --git a/libgui/graphics/QtHandlesUtils.cc b/libgui/graphics/QtHandlesUtils.cc
--- a/libgui/graphics/QtHandlesUtils.cc
+++ b/libgui/graphics/QtHandlesUtils.cc
@@ -215,18 +215,18 @@ Qt::Alignment fromHVAlign (const caseles
 }
 
 QImage makeImageFromCData (const octave_value& v, int width, int height)
 {
   dim_vector dv (v.dims ());
 
   if (dv.length () == 3 && dv(2) == 3)
     {
-      int w = qMin (dv(1), width);
-      int h = qMin (dv(0), height);
+      int w = qMin (dv(1), static_cast<octave_idx_type (width));
+      int h = qMin (dv(0), static_cast<octave_idx_type (height));
 
       int x_off = (w < width ? (width - w) / 2 : 0);
       int y_off = (h < height ? (height - h) / 2 : 0);
 
       QImage img (width, height, QImage::Format_ARGB32);
       img.fill (qRgba (0, 0, 0, 0));
 
       if (v.is_uint8_type ())
